drivers/mailbox/bcm2836/init.c

```diff
@@ -207,6 +207,20 @@ NTSTATUS RpiSetDeviceMacAddress (
         REG_SZ,
         &macAddrStr,
         sizeof(macAddrStr));
+    if (!NT_SUCCESS(status)) {
+        RPIQ_LOG_ERROR(
+            "Failed to set MAC value at NetworkAddress registry %!STATUS!",
+            status);
+        //goto End;
+    }
+	
+	    status = RtlWriteRegistryValue(
+        RTL_REGISTRY_CONTROL,
+        L"\\Class\\{4d36e972-e325-11ce-bfc1-08002be10318}\\0003",
```

Comment:
@dmcilvaney is this really safe to do? I thought the NIC value assignment in the registry is assigned on first-time boot in whatever order the NICs are enumerated (effectively random).
Any way around this?

Reply:
It is not reliable to assume the ethernet will enumerate to 0003. While it will **usually** be the same, additional external devices can change the enumeration. I am currently helping get a few RPi3b+ boards running for another team and they are enumerating to 0004.

A more sustainable approach might be to create a filter driver for both of the boards. The filter driver could call into the mailbox driver to get the correct value, and palace it in the correct location.